firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/ApiClient.java

```diff
@@ -89,13 +89,15 @@ static FetchEligibleCampaignsResponse createCacheExpiringResponse() {
     }
     Logging.logi(FETCHING_CAMPAIGN_MESSAGE);
     providerInstaller.install();
-    return firebaseInstanceId
-        .getInstanceId()
+    Task<String> idTask = firebaseInstallations.getId();
+    Task<InstallationTokenResult> tokenTask = firebaseInstallations.getToken(false);
+    return Tasks.whenAll(idTask, tokenTask)
         .continueWith(
-            instanceIdResultTask -> {
-              InstanceIdResult instanceIdResult = instanceIdResultTask.getResult();
-              if (instanceIdResult == null) {
-                Logging.logw("InstanceID is null, not calling backend");
+            (unused) -> {
+              String idResult = idTask.getResult();
+              InstallationTokenResult tokenResult = tokenTask.getResult();
+              if (tokenResult == null || idResult == null) {
```

Comment:
None of the tasks will return null. Its on me for not adding the annotations on the APIs. I will update that on my end.

The Tasks would throw exceptions which you might want to handle based on completion state.